[Metricbeat][Redis] Add Redis 6.0/7.0 INFO fields and improve field descriptions

giorgi-imerlishvili-elastic opened the pull request:
## Proposed commit message

TODO
<!-- Mandatory
Explain here the changes you made on the PR.

Please explain:

- WHAT: patterns used, algorithms implemented, design architecture, message processing, etc.
- WHY:  the rationale/motivation for the changes

This text will be pasted into the squash dialog when the change is committed and will be
a long term historical record of the change to help future contributors understand the
change, please help them by making it clear and comprehensive, they may be you.

If the commit title is adequate to describe both of these things, The text here may be omitted
or replaced with "See title". The title of the PR will be used as the commit message title when
the merge is made and the "See title" marker will be removed if present.

The text here and the PR title will be subject to the PR review process.
-->

## Checklist

<!-- Mandatory
Add a checklist of things that are required to be reviewed in order to have the PR approved

List here all the items you have verified BEFORE sending this PR. Please DO NOT remove any item, striking through those that do not apply. (Just in case, strikethrough uses two tildes. ~~Scratch this.~~)
-->

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works. Where relevant, I have used the [`stresstest.sh`](https://github.com/elastic/beats/blob/main/script/stresstest.sh) script to run them under stress conditions and race detector to verify their stability.
- [x] I have added an entry in `./changelog/fragments` using the [changelog tool](https://github.com/elastic/elastic-agent-changelog-tool/blob/main/docs/usage.md).

colleenmcginnis commented on metricbeat/module/redis/info/_meta/fields.yml:
```diff
@@ -83,12 +83,24 @@
           format: bytes
           description: >
             Used memory by the Lua engine.
+          deprecated: true
```
Can you use the syntax documented in [Contributing to the docs](https://www.elastic.co/docs/extend/beats/contributing-docs#update-fields)? Please change the version number as needed.

```suggestion
          version:
            deprecated: 9.3.0
```

colleenmcginnis commented on docs/reference/metricbeat/exported-fields-redis.md:
```diff
@@ -113,14 +113,22 @@ Redis memory stats.
 
 
-**`redis.info.memory.used.lua`**
+**`redis.info.memory.used.lua`** {applies_to}`stack: deprecated True`
```
After updating the syntax used in `metricbeat/module/redis/info/_meta/fields.yml` and rerunning `make update`, you should get something like this instead:

```suggestion
**`redis.info.memory.used.lua`** {applies_to}`stack: deprecated 9.3.0`
```

ishleenk17 commented on metricbeat/module/redis/info/_meta/fields.yml:
```diff
@@ -83,12 +83,25 @@
           description: >
             Used memory by the Lua engine.
+          version:
+           deprecated: 9.4.0
```
I have a question about how we're handling deprecated fields.
The deprecation is tied to Redis versions, not stack versions. For example, if someone is using stack version 9.4 with Redis 6.0, this field would show as deprecated — which would be incorrect in that scenario.

Am I understanding this correctly?

giorgi-imerlishvili-elastic replied:
> I have a question about how we're handling deprecated fields. The deprecation is tied to Redis versions, not stack versions. For example, if someone is using stack version 9.4 with Redis 6.0, this field would show as deprecated — which would be incorrect in that scenario.
> 
> Am I understanding this correctly?

@ishleenk17 I agree that if someone is using Redis 6.0 with the 9.4 stack, this might look odd.
Another option, I think, is that as long as our stack continues to support Redis 6.0 and older versions, we can keep this field without a deprecation notice. We would just add a comment specifying that in newer versions used_memory_vm_eval should be used instead, as @shmsr suggested in his patch.

ishleenk17 commented on docs/reference/metricbeat/exported-fields-redis.md:
```diff
@@ -747,6 +825,28 @@ Redis stats.
 
+## tracking [_tracking]
+
+Redis client-side caching tracking stats. Added in Redis 6.0.
```
We were already having support till 6.0, right ?
So that means these metrics were not added then. 
Lets see if all these metrics hold importance from custome rperspective.

giorgi-imerlishvili-elastic commented:
@shmsr 
> Yes please do test and see if all the metrics are getting collected properly. I'd suggest to test with Redis 6, 7, 8.

Did manual testing via metricbeat producing logs to elasticsearch and checked in Kibana logs. tested with redis:6.2.6-alpine, redis:7.4.7-alpine, redis:8.2.3-alpine images and no issues found

@ishleenk17
> I have a question about how we're handling deprecated fields.
> The deprecation is tied to Redis versions, not stack versions. For example, if someone is using stack version 9.4 with Redis > 6.0, this field would show as deprecated — which would be incorrect in that scenario.

Removed deprecation notice from used_memory_lua field. Because it deprecated since Redis 7.0 as part of this change we leave description that it's deprecated from Redis side and `used_memory_vm_eval` field should be used instead

> Added in Redis 7.0.
> I don't think we need to mention it in each field description when it was added.
> . We are mentioning it once under the VM category that should be good.

Removed repetitive comments under vm field

vishaangelova commented on metricbeat/module/redis/info/_meta/fields.yml:
```diff
@@ -82,12 +82,54 @@
+          type: long
+          format: bytes
+          description: >
+            Combined memory overhead from EVAL scripts and Functions (part of used_memory). Added in Redis 7.0.
```
```suggestion
            Combined memory overhead from EVAL scripts and Functions (part of `used_memory`). Added in Redis 7.0.
```

vishaangelova reviewed (approved):
Left some suggestions for your consideration, but otherwise the docs LGTM.
